static_src/components/overview_container.jsx

```diff
@@ -72,13 +72,18 @@ export default class OverviewContainer extends React.Component {
     );
 
     return (
-      <EntityEmpty callout="We can’t find any of your organizations.">
-        <p>
-          If you just joined, your organization may not yet be ready. Sometimes
-          organizations can take up to 20 minutes to appear on your first login.
-          { contactMsg }
-        </p>
-      </EntityEmpty>
+      <I18n>
+        {t => (
+          <EntityEmpty callout={t('We can’t find any of your organizations.')}>
```

Comment:
So, if the i18n library can't find a key, does it default to showing the provided string? I'm slightly confused about what the `I18n` component is doing here. Is it intended to be illustrative?

Reply:
Exactly - if the key does not exist it just shows the text per https://www.i18next.com/principles/fallback.html#key-not-found 

![image](https://user-images.githubusercontent.com/626664/31253271-ea5300be-aa6f-11e7-98c5-41af4b021d01.png)

WRT the `I18n` component: this is a component that provides a "render prop" per https://react.i18next.com/components/i18n-render-prop.html#i18n-render-prop

![image](https://user-images.githubusercontent.com/626664/31253321-291f1832-aa70-11e7-9bc7-92ace0ed96f5.png)

I'm not sure if you've followed Ryan Florence or Michael Jackson's recent discussions on HOCs versus render props, but what I've done here is used the render prop version instead of the HOC version. Here's their blog article and inside it is also the video on the talk that Michael Jackson gave: https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce. 

I'll let you dig into those links as you please, but essentially what I'm doing is including this `I18n` component which gives me access to `t` so that I can translate things. You have have noticed that I've used `<I18n />` deep within the render tree in certain places instead of the first node in the tree: this is because at this stage we're only translating certain specific strings (the "organization" -> "organisation" stuff). If we were actually providing 100% coverage of translating all text, then usually the `I18n` component would be the first node in the render tree because then all subsequent nodes would have access to `t`. Within a particular component you'd usually not have more than one rendering of `<I18n />` in a tree because you'd just refactor the multiple uses of `<I18n />` out to be at the top of the tree.

I hope that clears it up, let me know if you need more info.